fendermint/app/src/snapshot.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::BlockHeight;
+use fendermint_vm_interpreter::fvm::state::{FvmStateParams, Snapshot};
+use fvm_ipld_blockstore::Blockstore;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+/// The config params for snapshot
+#[derive(Clone)]
+pub struct SnapshotConfig {
+    path: String,
+    period: BlockHeight,
+}
+
+/// Manager the snapshot of the entire state tree and state params at certain blocks.
+#[derive(Clone)]
+pub struct SnapshotManager {
+    config: SnapshotConfig,
+    lock: Arc<AtomicBool>,
+}
+
+const STARTED: bool = true;
+
+impl SnapshotManager {
+    pub fn new(config: SnapshotConfig) -> Self {
+        Self {
+            config,
+            lock: Arc::new(AtomicBool::new(false)),
+        }
+    }
+
+    pub fn should_run(&self, height: BlockHeight) -> bool {
+        height % self.config.period == 0
+    }
```

Comment:
One issue with the way this is that it will create snapshots even while a node is trying to catch up with others by applying historical transactions. In that scenario this will just slow things down, so I'd recommend making this method asynchronous, and if the period says there should be a snapshot make, then first check with the Tendermint API what the sync status is, like [here](https://github.com/consensus-shipyard/fendermint/blob/main/fendermint/eth/api/src/apis/eth.rs#L649-L651), and only return `true` if this node is in sync with the others.

For that the `SnapshotManager` will need a Tendermint RPC client pointing at the current node, but we'll need this kind of thing for other purposes as well (broadcasting transactions), so this is natural.